src/common/test-transport.c

```diff
@@ -194,6 +195,53 @@ test_echo_and_close (TestCase *tc,
   WAIT_UNTIL (closed == TRUE);
 }
 
+static void
+test_ping_pong (void)
+{
+  JsonObject *reply = NULL;
+  MockTransport *mock;
+  CockpitTransport *transport;
+  GBytes *sent;
+
+  mock = mock_transport_new ();
+  transport = COCKPIT_TRANSPORT (mock);
+
+  sent = cockpit_transport_build_control ("command", "ping", "value", "blah", "other", "marmalade", NULL);
+  cockpit_transport_emit_recv (transport, NULL, sent);
+  g_bytes_unref (sent);
+
+  reply = mock_transport_pop_control (mock);
+
+  cockpit_assert_json_eq (reply, "{ \"command\": \"pong\", \"value\": \"blah\", \"other\": \"marmalade\" }");
```

Comment:
The ping above doesn't have a channel, but this does send a pong - so again it seems this corresponds to the documentation (always sends a pong) and thus be a (small) API break, but it sort of conotradicts the test case above.

Reply:
A "ping" without a channel (as with all control messages without a channel) is a single hop. This is expected. I've added a comment here.